### PR TITLE
feat: [PIE-4605]: UI changes for Triggers authentication

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 31 Issues 
* Code issue 31 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11654)

File: src/modules/30-secrets/components/SecretInput/SecretInput.tsx [hunk 1]
```diff
@@ -149,7 +149,7 @@ const SecretInput: React.FC<FormikSecretInput> = props => {
               className={css.containerLinkText}
             >
               <div>{secretReference ? getString('secrets.secret.configureSecret') : getPlaceHolder()}</div>
-              {secretReference ? <div>{`<${secretReference['name']}>`}</div> : null}
+              {secretReference && <div>{`<${secretReference['name'] ?? secretReference}>`}</div>}
```
Comment (AshwinBhatkal-Harness) [inline src/modules/30-secrets/components/SecretInput/SecretInput.tsx:152]: Do we need to use secretReference if it is not an object

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/utils.tsx [hunk 2]
```diff
@@ -229,11 +231,21 @@ export const getModifiedTemplateValues = (
...
+  getString: UseStringsReturn['getString'],
+  isWebhookGithubTriggerAuthentication = false
 ): ObjectSchema<Record<string, any> | undefined> => {
   return object().shape({
     name: NameSchema({ requiredErrorMsg: getString('triggers.validation.triggerName') }),
     identifier: IdentifierSchema(),
+    ...(isWebhookGithubTriggerAuthentication && {
+      encryptedWebhookSecretIdentifier: string().test(
+        getString('triggers.validation.configureSecret'),
+        getString('triggers.validation.configureSecret'),
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-triggers/components/Triggers/WebhookTrigger/utils.tsx:243]: Same values twice?

**Review by AshwinBhatkal-Harness — CHANGES_REQUESTED** [review]
Need to add copy confirmation modal after selecting secret

File: src/modules/72-triggers/components/Triggers/WebhookTrigger/utils.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/72-triggers/components/Triggers/WebhookTrigger/utils.tsx:243]: One is for name and another is for message. Updated the name ti the key

**Comment by pankajsingh-harness** [comment]
trigger cypress

**Comment by pankajsingh-harness** [comment]
retrigger coverage

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by sunnykesh-harness — DISMISSED** [review]

File: src/modules/30-secrets/components/SecretInput/SecretInput.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/30-secrets/components/SecretInput/SecretInput.tsx]: Removed this logic now.

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by sunnykesh-harness — DISMISSED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**